xray/pdf_utils.py

```diff
@@ -293,3 +293,25 @@ def get_bad_redactions(page: Page) -> list[RedactionType]:
     bad_redactions = filter_redactions_by_text(redactions)
     bad_redactions = filter_redactions_by_pixmap(bad_redactions, page)
     return bad_redactions
+
+
+def multi_colored(pixmap, threshold=128) -> bool:
+    """Check if a pixmap contains black and white pixels.
+
+    :param pixmap: The pixmap to check
+    :param threshold: The threshold to use for black and white pixels
+    :return true if black and white else false
+    """
```

Comment:
I think I'd prefer if this was `is_black_and_white`, without the threshold parameter, and if the threshold could be hardcoded with a comment explaining how it worked. 

If that's not a great idea, I'd love more documentation about what threshold means and why it works.

Reply:
we can drop the parameter and hard code it.  

I added this comment 

```    
Every pixel with an intensity below 128 is set to black,
and every pixel with an intensity of 128 or above is set to white

This allows us to convert gray scale [X][X] to black but otherwise retain
text.
```

I think its more about detecting light and dark pixels not black and white but assigns black or white 

so detect_light_and_dark ?